### PR TITLE
Schema: Build Task: Extract Schema + Schema Support in Scope

### PR DESCRIPTION
**Pull request by luvkapur**
## Proposed Changes

- Add Unimplemented Schema
- Parse Anonymous Default Function Export Schema
- Add Export Assignment Transformer
- Add Extract Schema Build Task
- Add Component Schema support in Scope; Fetch from Builder API

### CODE REVIEW

File: scopes/semantics/entities/semantic-schema/schemas/unimplemented-schema.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Location, SchemaNode } from '../schema-node';
+
+/**
+ * needed for better backward and forward compatibility.
```
Comment (davidfirst) [inline scopes/semantics/entities/semantic-schema/schemas/unimplemented-schema.ts:4]: the comment should be probably updated :)

File: scopes/semantics/schema/schema.task.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+export function getSchemaArtifactPath() {
+  return join(CAPSULE_ARTIFACTS_DIR, 'schema.json');
+}
+
+export function getSchemaArtifactDef() {
+  const def: ArtifactDefinition = {
+    name: SCHEMA_ARTIFACT_NAME,
+    rootDir: CAPSULE_ARTIFACTS_DIR,
+    globPatterns: ['*.json'],
```
Comment (GiladShoham) [inline scopes/semantics/schema/schema.task.ts:74]: You might get unrelated json files.
Either put a more specific glob, or nest your file inside a subfolder

File: scopes/semantics/schema/schema.main.runtime.ts [hunk 3]
```diff
@@ -77,12 +84,48 @@ export class SchemaMain {
    */
   async getSchema(component: Component): Promise<APISchema> {
     this.logger.debug(`getSchema of ${component.id.toString()}`);
-    const env = this.envs.getEnv(component).env;
-    if (typeof env.getSchemaExtractor === 'undefined') {
-      throw new Error(`No SchemaExtractor defined for ${env.name}`);
+
+    // if on workspace get schema from ts server
+    if (this.workspace) {
```
Comment (GiladShoham) [inline scopes/semantics/schema/schema.main.runtime.ts:89]: If this is not an expensive op it might be better to calculate it on load

**Review by GiladShoham — CHANGES_REQUESTED** [review]

File: scopes/semantics/schema/schema.task.ts (hunk 2 above)
Comment (luvkapur) [reply scopes/semantics/schema/schema.task.ts]: made it more specific `schema.json`

File: scopes/semantics/schema/schema.main.runtime.ts (hunk 3 above)
Comment (luvkapur) [reply scopes/semantics/schema/schema.main.runtime.ts:89]: it is as expensive as compiling the component
Not sure what happens onComponentLoad and where this should be persisted and then fetched from when eventually requested through the runtime's API
I do see some code commented out in the provider of the `SchemaMainRuntime` which tried to fetch the component schema. 
Will connect with David/Ran to understand it better

**Review by davidfirst — APPROVED** [review]

**Review by davidfirst — APPROVED** [review]
